search.go

```diff
@@ -128,6 +128,14 @@ func Range(queryPath QueryPath, begin Scalar, end Scalar) QueryPart {
 	}
 }
 
+// NotNil creates a query part where the value must exist.
+// This is done by finding results between byte 0x0 and 0xff
```

Comment:
`0xff` is actually just the last entry in the bucket? Since the keys are not 1-byte long? (well the might be, but probably not?)

Reply:
This is how the cursor works. It uses bytes.Compare so any value where the first byte is smaller than 0xff will match.